custom_components/ha_washdata/manager.py

```
@@ -2042,13 +2274,18 @@ async def _watchdog_check_stuck_cycle(self, now: datetime) -> None:
         # low_power_no_update_timeout is reached.
 
         # Dishwashers can have very long silent drying phases (up to 2h)
-        # We use the device-specific timeout as the floor for this effective timeout
+        # We use the device-specific timeout as the floor for this effective timeout.
+        # The floor is applied unconditionally — dishwashers have passive drying phases
+        # even when no profile has been matched yet.  The original restriction to matched
+        # cycles caused premature kills: with the default 3600s timeout, an unmatched
+        # dishwasher cycle was killed ~1h after the last sensor update, while the
+        # physical drying phase could still have 1–2h of silent runtime remaining.
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)